### ISSUE TITLE
Unexpected static keyword

### ISSUE THREAD

**Issue by roanutil** [nicklockwood/SwiftFormat#1461]
Running `swiftformat test.swift --rules redundantSelf` fails with `error: Unexpected static keyword on line 1` for the following code:

```swift
public struct FooClient: Sendable {
    public let mapFromBar: @Sendable (Bar) async -> Result<Foo, DecodingError>

    public let mapManyFromBars: @Sendable ([Bar]) async -> Result<[Foo], DecodingError>

    public let mapToBar: @Sendable (Foo) async -> Bar

    public let mapManyToBars: @Sendable ([Foo]) async -> [Bar]

    public init(
        mapFromBar: @escaping @Sendable (Bar) async -> Result<Foo, DecodingError>,
        mapManyFromBars: @escaping @Sendable ([Bar]) async -> Result<[Foo], DecodingError>,
        mapToBar: @escaping @Sendable (Foo) async -> Bar,
        mapManyToBars: @escaping @Sendable ([Foo]) async -> [Bar]
    ) {
        self.mapFromBar = mapFromBar
        self.mapManyFromBars = mapManyFromBars
        self.mapToBar = mapToBar
        self.mapManyToBars = mapManyToBars
    }

    public init(
        mapFromBar: @escaping @Sendable (Bar) async -> Result<Foo, DecodingError>,
        mapToBar: @escaping @Sendable (Foo) async -> Bar
    ) {
        self.mapFromBar = mapFromBar
        mapManyFromBars = Self.deriveMapFromMany(mapFromBar)
        self.mapToBar = mapToBar
        mapManyToBars = Self.deriveMapToMany(mapToBar)
    }

    public static func deriveMapFromMany(
        _ closure: @escaping @Sendable (Bar) async -> Result<Foo, DecodingError>
    ) -> @Sendable ([Bar]) async -> Result<[Foo], DecodingError> {
        { bars in
            var widgetBars = [Foo]()
            widgetBars.reserveCapacity(bars.count)
            for bar in bars {
                switch await closure(bar) {
                case let .success(widgetBar):
                    widgetBars.append(widgetBar)
                case let .failure(error):
                    return .failure(error)
                }
            }
            return .success(widgetBars)
        }
    }

    public static func deriveMapToMany(
        _ closure: @escaping @Sendable (Foo) async -> Bar
    ) -> @Sendable ([Foo]) async -> [Bar] {
        { foos in
            var bars = [Bar]()
            bars.reserveCapacity(foos.count)
            for foo in foos {
                let bar = await closure(foo)
                bars.append(bar)
            }
            return bars
        }
    }
}
```

swiftformat version = 0.51.10

Edit: Updated code sample

**Comment by nicklockwood** [comment]
@roanutil AFAIK it's illegal in swift to have a static function outside of a class, struct or enum, which is why SwiftFormat is complaining. If I try putting that same code inside a class it works OK.

**Comment by roanutil** [comment]
@nicklockwood Sorry, I messed up the code sample by trying to reduce it to a minimum. The updated sample should reproduce the problem.

**Comment by nicklockwood** [comment]
@roanutil thanks, I was able to reproduce it now 👍

**Comment by nicklockwood** [comment]
@roanutil fixed in 0.51.11
